cmd/eksctl-anywhere/cmd/createcluster.go

```diff
@@ -185,8 +193,45 @@ func (cc *createClusterOptions) createCluster(cmd *cobra.Command, _ []string) er
 		},
 		ManagementCluster: cluster,
 		Provider:          deps.Provider,
+		CliConfig:         cliConfig,
 	}
 	createValidations := createvalidations.New(validationOpts)
 
 	return createCluster.Run(ctx, clusterSpec, createValidations, cc.forceClean)
 }
+
+func (cc *createClusterOptions) directoriesToMount(clusterSpec *cluster.Spec, cliConfig *config.CliConfig) ([]string, error) {
+	dirs := cc.mountDirs()
+	fluxConfig := clusterSpec.FluxConfig
+	if fluxConfig == nil {
+		return dirs, nil
+	}
+	if fluxConfig.Spec.Git == nil {
+		return dirs, nil
+	}
+
+	if cliConfig.GitPrivateKeyFile != "" {
+		dirs = append(dirs, cliConfig.GitPrivateKeyFile)
+	}
+
+	return dirs, nil
+}
+
+func buildCliConfig(clusterSpec *cluster.Spec) *config.CliConfig {
+	if clusterSpec.FluxConfig == nil {
+		return nil
+	}
+	fluxConfig := clusterSpec.FluxConfig
+	if fluxConfig.Spec.Git == nil {
+		return nil
```

Comment:
I would suggest just returning an empty struct instead of nil
The code that needs to read this can decide what to do with empty fields
If we return nil here and the rest of the code relies on that, all that code need to change if we add more fields to this config